source/time_tables/departure_boards.cpp

```diff
@@ -621,14 +624,51 @@ void terminus_schedules(PbCreator& pb_creator,
         }
 
         map_route_stop_point[route_point] = stop_times;
+        if (map_route_route_point.find(route) == map_route_route_point.end()) {
+            map_route_route_point[route] = {route_point};
+        } else {
+            map_route_route_point[route].emplace_back(route_point);
+        }
     }
 
+    clean_terminus_schedules(map_route_route_point, response_status, map_route_stop_point);
+
     render(pb_creator, response_status, map_route_stop_point, map_route_point_first_last_st, handler.date_time,
            handler.max_datetime, calendar_id, depth);
 
     pb_creator.make_paginate(total_result, start_page, count, pb_creator.terminus_schedules_size());
 }
 
+void clean_terminus_schedules(const std::map<const type::Route*, vector_jpp_idx>& map_route_route_point,
+                              std::map<routing::JppIdx, pbnavitia::ResponseStatus>& response_status,
+                              std::map<routing::JppIdx, vector_dt_st>& map_route_stop_point) {
+    for (const auto& rt_rp : map_route_route_point) {
+        bool with_stop_times = false;
+        if (rt_rp.second.size() > 1) {
+            for (const auto& route_point : rt_rp.second) {
+                const auto& it = map_route_stop_point.find(route_point);
+                if (!it->second.empty()) {
```

Comment:
Will crash if `it == map_route_stop_point.end()`. Are you sure it cannot happen ?